src/config/protobuf.rs

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::io::Write;
+
+use tokio::sync::mpsc::{channel, Sender};
+use tonic::transport::Server;
+use tonic::{Code, Request, Response, Status};
+
+use crate::config::MessageHubOptions;
+use crate::proto_message_hub::astarte_message_hub_config_server::{
+    AstarteMessageHubConfig, AstarteMessageHubConfigServer,
+};
+use crate::proto_message_hub::ConfigMessage;
+
+#[derive(Debug)]
+struct AstarteMessageHubConfigImpl {
```

Comment:
Avoid to use the `Impl` suffix. Rename the proto service to `MessageHubConfig` and the struct to `AstarteMessageHubConfig`.